cmd/clusterctl/cmd/version_checker_test.go

```diff
@@ -123,6 +123,58 @@ New clusterctl version available: v0.3.8 -> v0.3.9-alpha.0
 https://github.com/foo/bar/releases/v0.3.9-alpha.0
 `,
 		},
+		{
+			name: "returns message if cli version is a pre-release of the latest available release",
+			cliVersion: func() version.Info {
+				return version.Info{
+					GitVersion: "v0.3.9-alpha.0",
+				}
+			},
+			githubResponse: `{"tag_name": "v0.3.9", "html_url": "https://github.com/foo/bar/releases/v0.3.9"}`,
+			expectErr:      false,
+			expectedOutput: `
+New clusterctl version available: v0.3.9-alpha.0 -> v0.3.9
+https://github.com/foo/bar/releases/v0.3.9
+`,
+		},
+		{
+			name: "returns message if cli version is a pre-release and latest available release is the next pre-release",
+			cliVersion: func() version.Info {
+				return version.Info{
+					GitVersion: "v0.3.8-alpha.0",
+				}
+			},
+			githubResponse: `{"tag_name": "v0.3.8-alpha.1", "html_url": "https://github.com/foo/bar/releases/v0.3.8-alpha.1"}`,
+			expectErr:      false,
+			expectedOutput: `
+New clusterctl version available: v0.3.8-alpha.0 -> v0.3.8-alpha.1
+https://github.com/foo/bar/releases/v0.3.8-alpha.1
+`,
```

Comment:
Same here. I wanted clusterctl to inform users if there were valid pre-releases available. 
As long as we don't name our pre-releases in the same format as `GitVersion` we should be good.